src/core/src/pass/serialize.cpp

```diff
@@ -456,11 +456,7 @@ class XmlSerializer : public ngraph::AttributeVisitor {
             m_xml_node.append_attribute(name.c_str()).set_value(join(attrs).c_str());
         } else if (const auto& a = ngraph::as_type<ngraph::AttributeAdapter<ov::PartialShape>>(&adapter)) {
             const auto& attrs = a->get();
-            std::stringstream shape_str_stream;
-            shape_str_stream << attrs;
-            auto shape_str = shape_str_stream.str();
-            if (shape_str[0] == '{' && shape_str[shape_str.size() - 1] == '}')
-                shape_str = shape_str.substr(1, shape_str.size() - 2);
+            auto shape_str = attrs.to_string();
```

Comment:
@popovaan Looks like before we had:
```
{1, 2, 3}
but remove {} and write only 1, 2, 3
```
Now you get:
```
[1, 2, 3]
but don't skip []
```
is it expected?

Reply:
I think having [] is good. It aligns with how ndarray.shape is printed in python

Reply:
Yes, it is expected. The part of this alignment to make PartialShape be sterilized with square brackets in IR.

Reply:
Will it preserve backward-compatibility?

Reply:
Yes, PartialShape() from string is used while deserializing and it accepts old format without square brackets.  
Check lines src/core/src/partial_shape.cpp in lines 30-31.

Reply:
Yes, IR will have shapes with square brackets for Parameter nodes.

Reply:
Discussed it with @ilyachur, this change makes impossible to read new IR in old openvino. So returned serializing with old format.